lib/oauth2/client.rb

```diff
@@ -158,17 +158,15 @@ def get_token(params, access_token_opts = {}, access_token_class = AccessToken)
       end
       opts[:headers].merge!(headers)
       response = request(options[:token_method], token_url, opts)
-      response_contains_token = response.parsed.is_a?(Hash) &&
-                                (response.parsed['access_token'] || response.parsed['id_token'])
 
-      if options[:raise_errors] && !response_contains_token
+      access_token = build_access_token(response, access_token_opts, access_token_class)
```

Comment:
👍 Determine response_contains_token or token valid should be token's responsible.